### ISSUE TITLE
Test Pilot social sharing messages for Facebook and Twitter are not localized in German

### ISSUE THREAD

**Issue by SoftVision-PaulOiegas** [mozilla/testpilot#2001]
[Affected versions]:
- Firefox 50.1.0 and up

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisite]:
- German language was added on the top of the list from Firefox "Options > Content > Languages" section or a localized DE build is used.
- Facebook and Twitter accounts are logged in.

[Steps to reproduce]:
1. Open Firefox browser and navigate to "https://testpilot.dev.mozaws.net/share/" page.
2. Click on Facebook and Twitter sharing buttons.
3. Share and tweet the auto generated message about Test Pilot.
4. Open Facebook and Twitter profile pages.
5. Observe the shared / tweeted message about Test Pilot.

[Expected result]:
- The auto generated message is shared as localized in German.

[Actual result]:
- The auto generated message is shared in English.

**Comment by lmorchard** [comment]
As far as I can tell, the only way to have localized Twitter & Facebook sharing messages is to share localized static page URLs. (i.e. `/en-US/experiments/foo` vs `/de/experiments/foo`) 

That's because the spiders for both services fetch the information from HTML `<meta>` headers - and they *do not run JavaScript*. So, our client-side localization will not get applied. On top of that, they tend to cache the scraped content for a long time per URL. So, we need server-side localization per-URL - which in our current stack means [static page generation](https://github.com/mozilla/testpilot/issues/1306) for each locale.

See also:
* https://twittercommunity.com/t/localization-of-twitter-cards/22249

**Comment by lmorchard** [comment]
There *does* appear to be a potential mechanism to use with Facebook, though:

* https://developers.facebook.com/docs/opengraph/guides/internationalization#objects

TL;DR: Facebook will look for `og:locale:alternate` meta tags like so: 
`<meta property="og:locale:alternate" content="fr_FR" />`

Then, the scraper will fetch that page for each locale listed with a `?fb_locale=fr_FR` parameter and look for localized meta tags. Still, this probably means we need some fancy coding in Nginx to handle the parameter and bounce to a statically generated page anyway.

**Comment by chuckharmston** [comment]
Given the infrequency with which this is used (less than 150 times ever, many of which will be SoftVision doing testing), the size of our non-English audience, and the amount of work it'd take to get this done correctly, I'm going to close it.
